t/14-grutasks.t

```diff
@@ -453,8 +457,13 @@ subtest 'Gru tasks TTL' => sub {
       or diag explain $result;
 
     my @ids;
-    for (1 .. 100) {
-        push @ids, $t->app->gru->enqueue(limit_assets => [] => {priority => 10, ttl => -50})->{minion_id};
+    # use one minion for test coverage collection but the rest without for
+    # performance reasons
+    my %data = (priority => 10, ttl => -50);
+    push @ids, $t->app->gru->enqueue(limit_assets => [] => \%data)->{minion_id};
+    %data = (%data, (notes => {no_cover => 1}));
```

Comment:
This looks wrong. The assignment of `%data` does *not* mean that this hash is destroyed and a new one is created. It means that the existing `%data` is mutated to have the same value as the temporary object specified on the right hand side of the assignment. That means the reference to `%data` passed one line before this assignment is affected as well. However, that's likely what you wanted to avoid here, right?

Reply:
Just to be sure I've just did the following test:

```
perl -de1
[...]
main::(-e:1):   1
  DB<1> %data = (prio => 10, ttl => -50);

  DB<2> use Data::Dumper;

  DB<3> print(Dumper(\%data));
$VAR1 = {
          'ttl' => -50,
          'prio' => 10
        };

  DB<4> $ref = \%data;

  DB<5> print(Dumper($ref));
$VAR1 = {
          'ttl' => -50,
          'prio' => 10
        };

  DB<6> %data = (%data, (notes => {no_cover => 1}));

  DB<7> print(Dumper($ref));
$VAR1 = {
          'notes' => {
                       'no_cover' => 1
                     },
          'prio' => 10,
          'ttl' => -50
        };
```

As you can see in <7>, the object `$ref` points to is affected by the <6>.

Reply:
You are correct that `%data` is mutated, but it should already be serialized to JSON and stored in the database by then.

Reply:
Exactly. I was aware that the existing object is mutated over the reference but assume it was used already in before I don't care.

Reply:
updated together with mkittler to make it explicit with `%data_no_cover`